src/components/FileTree/File.tsx

```diff
@@ -79,21 +79,16 @@ function EditableFile({
   setEditMode,
 }: FileProps & { editMode: boolean; setEditMode: (value: boolean) => void }) {
   const linkRef = useRef<HTMLAnchorElement>(null)
+  const hasEllipsis = useOverflowCheck(linkRef)
+
   const { mutateAsync: renameFile } = useRenameFile(file)
 
-  const navigate = useNavigate()
-
-  const hasEllipsis = useOverflowCheck(linkRef)
-  const fileExtension = file.fileName.split('.').pop()
+  const fileExtension = getFileExtension(file.fileName)
 
   const handleSave = async (newValue: string) => {
-    const newFileName = `${newValue}.${fileExtension}`
+    const newFileName = `${newValue.trim()}.${fileExtension}`
```

Comment:
Without this, it's easy to have multiple files with different names that look the same (e.g. `generator.k6g` and `generator .k6g`)